sar_pipeline/pipelines/isce3_rtc/cli.py

```diff
@@ -884,3 +909,203 @@ def make_rtc_opera_stac_and_upload_bursts(
+        os.makedirs(local_product_folder_1, exist_ok=True)
+        S3Downloader.download_folder(
+            s3_bucket, s3_product_folder_1, local_product_folder_1
+        )
+    if s3_product_folder_2:
+        # download product folder to local folder
+        local_product_folder_2 = out_folder / "folder_2" / s3_product_folder_2
+        os.makedirs(local_product_folder_2, exist_ok=True)
+        S3Downloader.download_folder(
+            s3_bucket, s3_product_folder_2, local_product_folder_2
+        )
+
+    # compare count of files and filetypes in folder
+    files_are_different, file_differences = compare_product_folder_files(
+        local_product_folder_1, local_product_folder_2
+    )
+    file_differences_path = out_folder / "file_differences.json"
+    if files_are_different:
+        logging.warning(
+            f"Differences in product files found. Saving summary to : {file_differences_path}"
+        )
+    else:
+        logger.info(
+            f"No differences in product files identified. Saving summary to : {file_differences_path}"
+        )
+    write_diffs_to_json(file_differences, file_differences_path)
+
+    # compare the json files
+    json_files_1 = list(local_product_folder_1.glob("*.json"))
+    json_files_2 = list(local_product_folder_2.glob("*.json"))
+    if len(json_files_1) > 1 or len(json_files_2) > 1:
+        logging.warning("multiple jsons found, comparing first")
+
+    json_differences = compare_json(str(json_files_1[0]), str(json_files_2[0]))
+    json_difference_path = out_folder / "json_differences.json"
+    if len(json_differences) > 0:
+        logging.warning(
+            f"Differences in json metadata found. Saving summary to : {json_difference_path}"
+        )
+    else:
+        logger.info(
+            f"No differences in json file identified. Saving summary to : {json_difference_path}"
+        )
+    write_diffs_to_json(json_differences, json_difference_path)
+
+    if product == "RTC_S1":
+        # compare the xml files - xml only exits for rtc_s1
+        xml_files_1 = list(local_product_folder_1.glob("*.xml"))
+        xml_files_2 = list(local_product_folder_2.glob("*.xml"))
+        if len(xml_files_1) > 1 or len(xml_files_1) > 1:
+            logging.warning("multiple xmls found, comparing first")
+
+        xml_differences = compare_xml(str(xml_files_1[0]), str(xml_files_2[0]))
+        xml_difference_path = out_folder / "xml_differences.xml"
+        if len(xml_differences) > 0:
+
+            logging.warning(
+                f"Differences in xml metadata found. Saving summary to : {xml_difference_path}"
+            )
+        else:
+            logger.info(
+                f"No differences in XML file identified. Saving summary to : {xml_difference_path}"
+            )
+        write_diffs_to_xml(xml_differences, xml_difference_path)
+
+    # compare the like tifs. e.g. compare mask to mask, HH-gamma0 to HH-gamma0
```

Comment:
I suggest adding an extra comment to show our convention (or example) for what the tifs look like e.g. OPERA_L2_RTC-S1_T007-014545-IW2_20250129T050922Z_20250703T034833Z_S1A_20_v1.0.4_mask.tif